subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformValuesInjectionIntegrationTest.groovy

```diff
@@ -211,15 +217,17 @@ project(':b') {
     }
 }
 
-abstract class MakeGreen extends ArtifactTransform {
+abstract class MakeGreen implements ArtifactTransformAction {
     @PrimaryInputDependencies
     abstract ${targetType} getDependencies()
+    @PrimaryInput
+    abstract File getInput()
     
-    List<File> transform(File input) {
+    void transform(ArtifactTransformOutputs outputs) {
         println "received dependencies files \${dependencies*.name} for processing \${input.name}"
-        def output = new File(outputDirectory, input.name + ".green")
+        def output = new File(outputs.workspace, input.name + ".green")
         output.text = "ok"
-        return [output]
+        outputs.registerOutput(output)
```

Comment:
Yeah, I'm quite sure we should have `registerOutput(String relativePath)` instead.